gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -147,11 +156,18 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
         datasetWorkUnitStates.iterator().next().getProp(CopySource.SERIALIZED_COPYABLE_DATASET));
     Path datasetWriterOutputPath = new Path(this.writerOutputDir, datasetAndPartition.identifier());
 
-    log.info(String.format("Publishing fileSet from %s for dataset %s", datasetWriterOutputPath,
-        metadata.getDatasetURN()));
+    log.info(String.format("[%s] Publishing fileSet from %s for dataset %s", datasetAndPartition.identifier(),
+        datasetWriterOutputPath, metadata.getDatasetURN()));
 
+    List<CommitStep> prePublish = getCommitSequence(datasetWorkUnitStates, PrePublishStep.class);
+    List<CommitStep> postPublish = getCommitSequence(datasetWorkUnitStates, PostPublishStep.class);
+    log.info(String.format("[%s] Found %d prePublish steps and %d postPublish steps.", datasetAndPartition.identifier(),
+        prePublish.size(), postPublish.size()));
+
+    executeCommitSequence(prePublish);
```

Comment:
What is the consequence of failing to execute these steps? Is it going to cause loss of data? Duplicate data? Or no problem?

Since commit steps and commit sequences were created for exactly-once semantics (at least the javadoc says so), if you are using them for different purposes can you either add some comments or modify the javadoc of commit step and commit seuqence.